langchain/document_loaders/json_loader.py

```diff
@@ -0,0 +1,97 @@
+"""Loader that loads data from JSON."""
+import json
+from pathlib import Path
+from typing import Callable, Dict, List, Union
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+
+class JSONLoader(BaseLoader):
+    """Loads a JSON file and references a jq schema provided to load the text into
+    documents.
+
+    Example:
+        [{"text": ...}, {"text": ...}, {"text": ...}] -> schema = .[].text
+        {"key": [{"text": ...}, {"text": ...}, {"text": ...}]} -> schema = .key[].text
+        ["", "", ""] -> schema = .[]
+    """
+
+    def __init__(
+        self,
+        file_path: Union[str, Path],
+        jq_schema: str,
+        content_key: str = None,
+        metadata_func: Callable[Dict, Dict] = None,
+    ):
+        """Initialize the JSONLoader.
+
+        Args:
+            file_path (Union[str, Path]): The path to the JSON file.
+            jq_schema (str): The jq schema to use to extract the data or text from the JSON.
+            content_key (str): The key to use to extract the content from the JSON if the jq_schema results to a list of objects (dict).
+            metadata_func (Callable[Dict, Dict]): A function that takes in the JSON object extracted by the jq_schema and the default metadata and returns a dict of the updated metadata.
+        """
+        try:
+            import jq  # noqa:F401
+        except ImportError:
+            raise ValueError(
+                "jq package not found, please install it with " "`pipenv install jq`"
```

Comment:
nit: combine strings and would just do pip (instead of pipenv, which is bit less universal)

Reply:
Oh, right! I also had to fix some linting issues so I just included this change in the latest commit. Thanks!